connections.go

```diff
@@ -65,6 +68,7 @@ type Connection struct {
 	Facility     *Facility            `json:"facility,omitempty"`
 	Metro        *Metro               `json:"metro,omitempty"`
 	Type         ConnectionType       `json:"type,omitempty"`
+	Mode         ConnectionMode       `json:"mode,omitempty"`
```

Comment:
When Type is `shared`, the Mode will not be included in the response (or may be `null`). 
I think we should try to represent that with a `*ConnectionMode` as we do for other null'able response properties.